linux-kernel (Linux Kernel): update to 6.10

MingcongBai opened the pull request:
Topic Description
-----------------

- kernel-tools: update to 6.10.0
    - Limit Hyper-V tools to AMD64, AArch64.
    - Rename hv_fcopy_daemon => hv_fcopy_uio_daemon.
- linux+kernel: update to 6.10.0
- linux-kernel: update to 6.10.0
    - Track patches at AOSC-Tracking/linux @ aosc/v6.10. [^1]
    - Backport pre-upstream LoongArch KVM LBT extension support. [^2]
    - Backport pre-upstream LoongArch KVM steal time support. [^3]
    - Backport pre-upstream LoongArch PWM support (v4). [^4]
    - Backport pre-upstream LoongArch CPUFreq support. [^5]
    - Backport pre-upstream LoongArch LIOINTC (Legacy I/O Interrupt Controller)
    ISR (interrupt service routine) fix. [^6]
    - Backport complete ntsync5 patchset (v4). [^7]
    - Backport pre-upstream Loongson DWMAC (7a13) patchset v13. [^8]
    - (all) Refresh configuration, enable new features - items of note:
       - Microsoft Azure Network Adapter
       - Realtek 8723CS (SDIO) and 8922AE
       - NT synchronization primitive driver (ntsync5)
    - (amd64) Enable new features - items of note:
       - Dell AIO (Optiplex 5400) UART Backlight
       - Intel IPU6 interface (improved support for >= 12th Gen laptop webcams)
       - Lenovo ThinkStation sensor reader
       - Lenovo Yoga Tablet 2 1380 fast charging support
       - Lenovo WMI Camera Button
       - Zhaoxin I2C Interface
    - (arm64) Enable new features - items of note:
       - Arm Trusted Services TEE driver
       - Acer Aspire 1 Embedded Controller
       - Panthor (DRM support for ARM Mali CSF-based GPUs)
       - Rockchip RK3308 audio CODEC
       - Rockchip USBDP COMBO PHY Driver
       - Tegra Security Engine
    - (loongarch64) Enable new features - items of note:
       - Gate upstream amdgpu workaround for cache flush issue, which is known
       to cause instability (GPU resets) on GCN 1.0 - 4.0
       - Drop DPM (dynamic power management) disabling hack for Polaris cards
       (gfx8, GCN 4th generation)
       - CPU frequency scaling support
       - Para-virtualization support
    - (loongson3) Enable new features - items of note:
       - Gate upstream amdgpu workaround for cache flush issue, which is known
       to cause instability (GPU resets) on GCN 1.0 - 4.0
    - (riscv64) Enable new features - items of note:
       - Allwinner D1 internal LDOs
       - Sophgo CV1800 series clock controller
    [^1]: https://github.com/AOSC-Tracking/linux/tree/aosc/v6.10
    [^2]: https://lore.kernel.org/loongarch/20240527074644.836699-1-maobibo@loongson.cn/T/#t
    [^3]: https://lore.kernel.org/loongarch/20240524073812.731032-1-maobibo@loongson.cn/T/#t
    [^4]: https://lore.kernel.org/loongarch/CAMpQs4L=JV_tSQN_LgcwzAUjKvQ-vt8zcJO9OaKQha6zCztw6w@mail.gmail.com/T/#t
    [^5]: https://lore.kernel.org/loongarch/20240612064205.2041548-1-chenhuacai@loongson.cn/T/#t
    [^6]: https://lore.kernel.org/loongarch/20240612070106.2060334-1-chenhuacai@loongson.cn/T/#u
    [^7]: https://lore.kernel.org/linux-kernel/20240416010837.333694-10-zfigura@codeweavers.com/T/#u
    [^8]: https://patchwork.kernel.org/project/netdevbpf/cover/cover.1716973237.git.siyanteng@loongson.cn/

Package(s) Affected
-------------------

- kernel-tools: 6.10.0-1
- linux+kernel: 3:6.10.0-1
- linux-kernel-6.10.0: 1:6.10.0-1

Security Update?
----------------

No

Build Order
-----------

```
#buildit linux-kernel linux+kernel kernel-tools
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

**Experimental Architectures**

- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`

MingcongBai committed:
linux-kernel: update to 6.10.0

- Track patches at AOSC-Tracking/linux @ aosc/v6.10. [^1]
- Backport pre-upstream LoongArch KVM LBT extension support. [^2]
- Backport pre-upstream LoongArch KVM steal time support. [^3]
- Backport pre-upstream LoongArch PWM support (v4). [^4]
- Backport pre-upstream LoongArch CPUFreq support. [^5]
- Backport pre-upstream LoongArch LIOINTC (Legacy I/O Interrupt Controller)
  ISR (interrupt service routine) fix. [^6]
- Backport complete ntsync5 patchset (v4). [^7]
- Backport pre-upstream Loongson DWMAC (7a13) patchset v13. [^8]
- (all) Refresh configuration, enable new features - items of note:
    - Microsoft Azure Network Adapter
    - Realtek 8723CS (SDIO) and 8922AE
    - NT synchronization primitive driver (ntsync5)
- (amd64) Enable new features - items of note:
    - Dell AIO (Optiplex 5400) UART Backlight
    - Intel IPU6 interface (improved support for >= 12th Gen laptop webcams)
    - Lenovo ThinkStation sensor reader
    - Lenovo Yoga Tablet 2 1380 fast charging support
    - Lenovo WMI Camera Button
    - Zhaoxin I2C Interface
- (arm64) Enable new features - items of note:
    - Arm Trusted Services TEE driver
    - Acer Aspire 1 Embedded Controller
    - Panthor (DRM support for ARM Mali CSF-based GPUs)
    - Rockchip RK3308 audio CODEC
    - Rockchip USBDP COMBO PHY Driver
    - Tegra Security Engine
- (loongarch64) Enable new features - items of note:
    - Gate upstream amdgpu workaround for cache flush issue, which is known
      to cause instability (GPU resets) on GCN 1.0 - 4.0
    - Drop DPM (dynamic power management) disabling hack for Polaris cards
      (gfx8, GCN 4th generation)
    - CPU frequency scaling support
    - Para-virtualization support
- (loongson3) Enable new features - items of note:
    - Gate upstream amdgpu workaround for cache flush issue, which is known
      to cause instability (GPU resets) on GCN 1.0 - 4.0
- (riscv64) Enable new features - items of note:
    - Allwinner D1 internal LDOs
    - Sophgo CV1800 series clock controller

[^1]: https://github.com/AOSC-Tracking/linux/tree/aosc/v6.10
[^2]: https://lore.kernel.org/loongarch/20240527074644.836699-1-maobibo@loongson.cn/T/#t
[^3]: https://lore.kernel.org/loongarch/20240524073812.731032-1-maobibo@loongson.cn/T/#t
[^4]: https://lore.kernel.org/loongarch/CAMpQs4L=JV_tSQN_LgcwzAUjKvQ-vt8zcJO9OaKQha6zCztw6w@mail.gmail.com/T/#t
[^5]: https://lore.kernel.org/loongarch/20240612064205.2041548-1-chenhuacai@loongson.cn/T/#t
[^6]: https://lore.kernel.org/loongarch/20240612070106.2060334-1-chenhuacai@loongson.cn/T/#u
[^7]: https://lore.kernel.org/linux-kernel/20240416010837.333694-10-zfigura@codeweavers.com/T/#u
[^8]: https://patchwork.kernel.org/project/netdevbpf/cover/cover.1716973237.git.siyanteng@loongson.cn/

MingcongBai committed:
linux+kernel: update to 6.10.0

OriginCode commented:
LGTM on amd64:
![圖片](https://github.com/AOSC-Dev/aosc-os-abbs/assets/17294071/75d4b997-f6dd-48af-8931-751df6060512)

MingcongBai committed:
kernel-tools: update to 6.10.0

- Limit Hyper-V tools to AMD64, AArch64.
- Rename hv_fcopy_daemon => hv_fcopy_uio_daemon.

KexyBiscuit commented:
Aren't Hyper-V tools also needed on Arm64?

KexyBiscuit commented:
`hv_kvp_daemon` and `hv_vss_daemon` are available on AArch64, and `hv_fcopy_uio_daemon` is only available on x86-{64,32}.

https://github.com/torvalds/linux/blob/b1bc554e009e3aeed7e4cfd2e717c7a34a98c683/tools/hv/Makefile#L23

KexyBiscuit commented on app-admin/kernel-tools/autobuild/build:
```diff
@@ -109,16 +109,18 @@ if ab_match_arch amd64; then
-        "$PKGDIR"/usr/bin/$i
-done
-install -dvm755 "$PKGDIR"/usr/lib/hyperv/kvp_scripts
+if ab_match_arch amd64; then
```
https://github.com/AOSC-Dev/aosc-os-abbs/pull/6352#issuecomment-2236847517

KexyBiscuit commented on app-admin/kernel-tools/autobuild/build.stage2:
```diff
@@ -92,16 +92,18 @@ if ab_match_arch amd64; then
-        "$PKGDIR"/usr/bin/$i
-done
-install -dvm755 "$PKGDIR"/usr/lib/hyperv/kvp_scripts
+if ab_match_arch amd64; then
```
https://github.com/AOSC-Dev/aosc-os-abbs/pull/6352#issuecomment-2236847517

KexyBiscuit commented on app-admin/kernel-tools/spec:
```diff
@@ -1,8 +1,12 @@
+# To RFC: Increment by 0.1 per RC release, i.e., RC3 = 0.3.
+#REL=0.${RC}
+REL=1
 #SRCS="tbl::https://git.kernel.org/torvalds/t/linux-${VER%%.0}-rc${RC}.tar.gz"
```
gz? xz?

MingcongBai replied:
RC tarballs are only available in .tar.gz.

KexyBiscuit commented on runtime-kernel/linux-kernel/spec:
```diff
@@ -1,8 +1,12 @@
+# To RFC: Increment by 0.1 per RC release, i.e., RC3 = 0.3.
+#REL=0.${RC}
+REL=1
 #SRCS="tbl::https://git.kernel.org/torvalds/t/linux-${VER%%.0}-rc${RC}.tar.gz"
```
gz? xz?

MingcongBai replied:
Ditto above about RC tarballs.

Cyanoxygen commented:
Testing
--------

Tested on amd64, with NVIDIA driver 555.58:

![image](https://github.com/user-attachments/assets/e53da999-a005-423e-b0c8-d9cf58b0ebb3)
